### ISSUE TITLE
Only obs processing in AeroVal

### ISSUE THREAD

**Issue by jgliss** [metno/pyaerocom#436]
Since the new trends interface is now also handled via the AeroVal interface (and pyaerocom backend processing), we need some functionality to handle a configuration setup (`EvalSetup`) such that it can process an experiment that contains only observations. In other words:

Suppose you want to run an experiment specified in an instance of `CFG=pya.aeroval.setupclasses.EvalSetup` via 
```python 
import pyaerocom as pya
processor = pya.aeroval.ExperimentProcessor(CFG) #with processor.cfg.model_cfg being empty.
processor.run()
```

Since AeroVal processing all goes via co-location and `ColocatedData` objects, we need a fake model to be used, thus, I suggest the following:

**On the fly generation of dummy model data in AeroCom format in a temporary location and add these dummy NetCDF files to `EvalSetup.model_cfg` (will be the only entry).** 

Steps:

1. In `processor.run()` check if `processor.cfg.model_cfg` is empty. If so, then:
2. Create a suitable temporary output location for the data, e.g. `tmpdir=pya.const.LOCAL_TMP_DIR` (which should be ~/MyPyaerocom/tmp
3. Register location of `tmpdir` in pyaerocom as search directory (so the dummy output can be found later on): `pya.const.add_data_search_dir(tmpdir)`.
4. Create dummy model output folder (and subfolder "renamed"), e.g.: `outdir="~/MyPyaerocom/tmp/dummy_model/renamed"` (**dummy_model** will be the model ID to be used).
5. For each obs variable (derived `EvalSetup.obs_cfg`) create a dummy iris Cube (called `cube` below) with low lat/lon resolution covering the whole globe (can be very low res, even only 2x2 grid points) and the timestamps according to the desired period and frequency specified in `EvalSetup`: should cover the total time span required based on the defined periods in the highest specified output resolution (e.g. 2000-2020, daily). Make sure the dummy cube has the correct variable name (`var_name`) and units (see pyaerocom/data/variables.ini for default units for each variable), `ts_type`, and var_name are assigned correctly).
6. Convert that `cube` to `GriddedData`: `dummy_vardata=pya.GriddedData(cube)`
7. Since AeroCom convention demands single year files: Split `dummy_vardata` into `GriddedData` into single year objects (`dummy_vardata_yr`) and for each of these single year GriddedData objects:
8. Save the dummy data in the output directory: `dummy_vardata_yr.to_netcdf(outdir)` (filename will be based on AeroCom convention).
9. Repeat steps 5-8 for each observation variable.
10. If all this worked correctly you should be able to find and read the dummy data like any other model (you may try via `print(pya.io.ReadGridded(model_id="dummy_model")` and check if it is found and all variables, years, frequencies are correct.
11. Add dummy model to `processor.cfg.model_cfg`, e.g.:  `processor.cfg.model_cfg['dummy'] = pya.aeroval.modelentry.ModelEntry(model_id='dummy_model')`
12. This should be sufficient for the processing to run through for all variables.
13. Don't forget to delete the dummy data at the end of `processor.run().
14. Then, only @AugustinMortier needs to figure out a way to hide the dummy model output (perhaps this could also be handled more explicit by defining a new analysis option `obs-only` in one of the classes `EvalSetup` is composed of.

**Comment by jgliss** [comment]
@dulte here you go, this is as good of an explanation I could come up with for now. Not sure if this should go into the upcoming release of v0.12.0 which I would like to finish next week. Otherwise 0.12.1 or so.

**Comment by jgliss** [comment]
And [here](https://github.com/metno/pyaerocom/blob/2633a630d065cca4f5c1dd8e3d3294ee74e1e604/tests/_conftest_helpers.py#L409) an example that should help you with creating dummy cubes :)

**Comment by avaldebe** [comment]
> **On the fly generation of dummy model data in AeroCom format in a temporary location and add these dummy NetCDF files to `EvalSetup.model_cfg` (will be the only entry).**

I would like to point out that his is a temporary/interim solution (i.e., a hack). At some point we'll need to implement a calculation path that does not go trough `ColocatedData`.

**Comment by jgliss** [comment]
> > **On the fly generation of dummy model data in AeroCom format in a temporary location and add these dummy NetCDF files to `EvalSetup.model_cfg` (will be the only entry).**
> 
> I would like to point out that his is a temporary/interim solution (i.e., a hack). At some point we'll need to implement a calculation path that does not go trough `ColocatedData`.

Agreed, but since we need this rather quickly, this should do the job for now. A processing path that does not go via `ColocatedData` will be a rather big effort in terms of API changes and should be planned thoroughly (originally, AeroVal was supposed to be a model evaluation interface, i.e. something where different datasets are compared with each other, with these new requirements, however, it becomes also an interface for visualisation and evaluation of single datasets).

**Comment by AugustinMortier** [comment]
It would be nice to make some progress on this theme, as this is required for replacing the [old interface](https://aerocom-trends.met.no) for which I actually had some requests.

**Comment by dulte** [comment]
I think it works... First evaluation of obs-only can be found here: https://gitlab.met.no/aeroval/testing

**Comment by AugustinMortier** [comment]
Nice first results!
I think that we have everything now to reproduce the trends interface. Maybe this belongs to another issue then (if we have any issue).

![Screenshot from 2022-03-03 16-56-50](https://user-images.githubusercontent.com/19610522/156601807-7476fbc8-7bb2-4161-93a1-b0b7490d6139.png)
